planet/subscription_request.py

```diff
@@ -0,0 +1,447 @@
+
+    Raises:
+        planet.exceptions.ClientError: If aoi is not a valid polygon or
+            multipolygon.
+    '''
+    valid_types = ['Polygon', 'MultiPolygon']
+
+    geom = geojson.as_geom(aoi)
+    if geom['type'].lower() not in [v.lower() for v in valid_types]:
+        raise ClientError(
+            f'Invalid geometry type: {geom["type"]} is not in {valid_types}.')
+
+    return _tool('clip', {'aoi': aoi})
+
+
+def file_format_tool(file_format: str) -> dict:
+    '''Specify a subscriptions API file format tool.
+
+    Parameters:
+        file_format: The format of the tool output. Either "COG" or "PL_NITF".
+
+    Raises:
+        planet.exceptions.ClientError: If file_format is not valid.
+    '''
+    try:
+        file_format = specs.validate_file_format(file_format)
+    except specs.SpecificationException as e:
+        raise ClientError(e)
+
+    return _tool('file_format', {'format': file_format})
+
+
+def harmonize_tool(target_sensor: str) -> dict:
+    '''Specify a subscriptions API harmonize tool.
+
+    Each sensor value transforms items captured by a defined set of instrument
+    IDs. Items which have not been captured by that defined set of instrument
+    IDs are unaffected by (passed through) the harmonization operation.
+
+    Parameters:
+        target_sensor: A value indicating to what sensor the input asset types
+            should be calibrated.
+
+    Raises:
+        planet.exceptions.ClientError: If target_sensor is not valid.
+    '''
+    try:
+        target_sensor = specs.get_match(target_sensor,
+                                        specs.HARMONIZE_TOOL_TARGET_SENSORS,
+                                        'target_sensor')
+    except specs.SpecificationException as e:
+        raise ClientError(e)
+
+    return _tool('harmonize', {'target_sensor': target_sensor})
+
+
+def reproject_tool(projection: str,
+                   resolution: Optional[float] = None,
+                   kernel: str = REPROJECT_KERNEL_DEFAULT) -> dict:
+    '''Specify a subscriptions API reproject tool.
+
+    Parameters:
+        projection: A coordinate system in the form EPSG:n (for example,
+            EPSG:4326 for WGS84, EPSG:32611 for UTM 11 North (WGS84), or
+            EPSG:3857 for Web Mercator). Well known text CRS values are also
+            supported (for example, WGS84).
+        resolution: The pixel width and height in the output file. If not
+            provided, the default is the resolution of the input item. This
+            value is in meters unless the coordinate system is geographic (such
+            as EPSG:4326), in which case, it is pixel size in decimal degrees.
+        kernel: The resampling kernel used. UDM files always use "near".
+
+    Raises:
+        planet.exceptions.ClientError: If kernel is not valid.
+    '''
+    try:
+        kernel = specs.get_match(kernel, REPROJECT_KERNEL, 'kernel')
+    except specs.SpecificationException as e:
+        raise ClientError(e)
+
```

Comment:
I really like how `get_match()` returns the ClientError here. It would be helpful to do a validation of the projection, something like `specs.get_match(projection, PROJECTION_OPTIONS, 'projection')`.

Reply:
Unfortunately, projections differ from kernels in that there are SOOO many valid projections that we can't just create a set of valid ones to match against.